cypress/e2e/05-get_result_from_patient_link.cy.js

```diff
@@ -1,28 +1,85 @@
+  cleanUpRunOktaOrgs,
+  setupTestOrder,
+} = require("../utils/setup-utils");
+
+const specRunName = "spec05";
+const currentSpecRunVersionName = `${testNumber()}-cypress-${specRunName}`;
 
 loginHooks();
 describe("Getting a test result from a patient link", () => {
-  let patientLink, patientDOB, patientObfuscatedName;
-  before("retrieve the patient link and dob", () => {
-    cy.task("getTestEventId").then((testEventId) => {
-      getPatientLinkByTestEventId(testEventId).then((res) => {
-        const patientLinkId = res.body.data.testResult.patientLink.internalId;
+  const patient = generatePatient();
+  const covidOnlyDevice = generateCovidOnlyDevice();
+  const patientDOB = patient.dobForPatientLink;
+  const patientObfuscatedName =
+    patient.firstName + " " + patient.lastName[0] + ".";
+  let patientLink;
+
+  before("setup spec data", () => {
+    cy.task("getSpecRunVersionName", specRunName).then(
+      (prevSpecRunVersionName) => {
+        if (prevSpecRunVersionName) {
+          cleanUpPreviousRunSetupData(prevSpecRunVersionName);
+          cleanUpRunOktaOrgs(prevSpecRunVersionName);
+        }
+        let data = {
+          specRunName: specRunName,
+          versionName: currentSpecRunVersionName,
+        };
+        cy.task("setSpecRunVersionName", data);
+        let patientId, createdDeviceId, specimenTypeId, testEventId;
+        setupRunData(currentSpecRunVersionName)
+          .then(() => setupPatient(currentSpecRunVersionName, patient))
+          .then((result) => {
+            patientId = result.internalId;
+          })
+          .then(() =>
+            setupCovidOnlyDevice(currentSpecRunVersionName, covidOnlyDevice),
+          )
+          .then((result) => {
+            createdDeviceId = result.createdDeviceId;
+            specimenTypeId = result.specimenTypeId;
+          })
+          .then(() =>
+            setupTestOrder(
+              currentSpecRunVersionName,
+              patientId,
+              createdDeviceId,
+              specimenTypeId,
+            ),
+          )
+          .then((result) => {
+            testEventId = result.body.data.submitQueueItem.testEventId;
+          })
+          .then(() => getPatientLinkByTestEventId(testEventId))
+          .then((result) => {
+            const patientLinkId =
+              result.body.data.testResult.patientLink.internalId;
 
-        patientLink = `${frontendURL}pxp?plid=${patientLinkId}`;
-      });
-    });
+            patientLink = `${frontendURL}pxp?plid=${patientLinkId}`;
+          });
+      },
+    );
+  });
 
-    cy.task("getPatientDOB").then((dob) => {
-      patientDOB = dob;
-    });
-    cy.task("getPatientName").then((name) => {
-      const [lastName, firstName] = name.split(",");
-      patientObfuscatedName = firstName + " " + lastName[0] + ".";
-    });
+  after("clean up spec data", () => {
+    cleanUpPreviousRunSetupData(currentSpecRunVersionName);
+    cleanUpRunOktaOrgs(currentSpecRunVersionName);
```

Comment:
I think we have this in some of the other PR's (and have seen the behavior there as well), but I'm intermittently getting failures in the after all hook because the attempt to fetch the orgs in `cleanUpRunOktaOrgs` fails after we delete that org in `cleanUpPreviousRunSetupData`. Sometimes if you run the tests, the after hooks will pull up the "something went wrong" page because of the permissions issue. 

If we swap the order of these two though, I think that fixes it, since the Okta deletion method does a fetch also for deleted orgs. It doesn't seem to be flaking in CI (yet) but wondering if you can replicate? If yes, maybe worth a follow up PR to swap the order of all of these in the after hook (or better yet, just making a single cleanup function that all the specs can then call).

Reply:
I've seen that something went wrong page at the end of the spec run before but I haven't had it actually fail for the last test. I think after all these rerunnable PRs get merged we'll certainly do some cleanup and refactoring PRs that could fix it though